cloudinit/sources/DataSourceAzure.py

```diff
@@ -371,17 +377,15 @@ def _check_if_primary(self, ephipv4: EphemeralIPv4Network) -> bool:
             # Primary nics must contain routes.
             return False
 
-        routed_networks = [r[0] for r in ephipv4.static_routes]
-        wireserver_route = f"{self._wireserver_endpoint}/32"
-        primary = any(
-            n in routed_networks
-            for n in [
-                "169.254.169.254/32",
-                wireserver_route,
-            ]
+        routed_networks = [r[0].split("/")[0] for r in ephipv4.static_routes]
+        self._route_configured_for_imds = "169.254.169.254" in routed_networks
+        self._route_configured_for_wireserver = (
+            self._wireserver_endpoint in routed_networks
+        )
```

Comment:
Let's add a leading comment for future-us above each attribute here to represent the config we expect from the separate Azure product behaviors.
```suggestion
        # Expected for primary NIC on Azure Public cloud offerings
        self._route_configured_for_imds = "169.254.169.254" in routed_networks
        # Expected support for primary NIC on Azure Stack
        self._route_configured_for_wireserver = (
            self._wireserver_endpoint in routed_networks
        )
```

Reply:
FWIW _route_configured_for_wireserver = True is expected for all scenarios.  I'm largely tracking it for logging purposes in a manner consistent with imds route.  It would be helpful should we ever see a case where it is reported as false for unknown reasons (perhaps we DHCP'd on secondary interface or somehow DHCP response was incorrect).

I will add documentation to clarify, thanks!

Reply:
done!